awx/ui_next/src/screens/Job/JobOutput/JobOutput.jsx

```diff
@@ -1,13 +1,287 @@
+  }
+
+  componentDidMount() {
+    this.loadJobEvents();
+  }
+
+  componentDidUpdate(prevProps, prevState) {
+    // recompute row heights for any job events that have transitioned
+    // from loading to loaded
+    const { currentlyLoading } = this.state;
+    let shouldRecomputeRowHeights = false;
+    prevState.currentlyLoading
+      .filter(n => !currentlyLoading.includes(n))
+      .forEach(n => {
+        shouldRecomputeRowHeights = true;
+        this.cache.clear(n);
+      });
+    if (shouldRecomputeRowHeights) {
+      if (this.listRef.recomputeRowHeights) {
+        this.listRef.recomputeRowHeights();
+      }
+    }
+  }
+
+  async loadJobEvents() {
+    const { job } = this.props;
+
+    const loadRange = range(1, 50);
+    this.setState(({ currentlyLoading }) => ({
+      hasContentLoading: true,
+      currentlyLoading: currentlyLoading.concat(loadRange),
+    }));
+    try {
+      const {
+        data: { results: newResults = [], count },
+      } = await JobsAPI.readEvents(job.id, job.type, {
+        page_size: 50,
+        order_by: 'start_line',
+      });
+      this.setState(({ results }) => {
+        newResults.forEach(jobEvent => {
+          results[jobEvent.counter] = jobEvent;
+        });
+        return { results, remoteRowCount: count + 1 };
+      });
+    } catch (err) {
+      this.setState({ contentError: err });
+    } finally {
+      this.setState(({ currentlyLoading }) => ({
+        hasContentLoading: false,
+        currentlyLoading: currentlyLoading.filter(n => !loadRange.includes(n)),
+      }));
+    }
+  }
+
+  isRowLoaded({ index }) {
+    const { results, currentlyLoading } = this.state;
+    if (results[index]) {
+      return true;
+    }
+    return currentlyLoading.includes(index);
+  }
+
+  rowRenderer({ index, parent, key, style }) {
+    const { results } = this.state;
+    return (
+      <CellMeasurer
+        key={key}
+        cache={this.cache}
+        parent={parent}
+        rowIndex={index}
+        columnIndex={0}
+      >
+        {results[index] ? (
+          <JobEvent className="row" style={style} {...results[index]} />
+        ) : (
+          <JobEventSkeleton
+            className="row"
+            style={style}
+            counter={index}
+            contentLength={80}
+          />
+        )}
+      </CellMeasurer>
+    );
+  }
+
+  loadMoreRows({ startIndex, stopIndex }) {
+    if (startIndex === 0 && stopIndex === 0) {
+      return Promise.resolve(null);
+    }
+    const { job } = this.props;
+
+    const loadRange = range(startIndex, stopIndex);
+    this.setState(({ currentlyLoading }) => ({
+      currentlyLoading: currentlyLoading.concat(loadRange),
+    }));
+    const params = {
+      counter__gte: startIndex,
+      counter__lte: stopIndex,
+      order_by: 'start_line',
+    };
+
+    return JobsAPI.readEvents(job.id, job.type, params).then(response => {
+      this.setState(({ results, currentlyLoading }) => {
```

Comment:
I get a console warning when I navigate away from the page: `Warning: Can't perform a React state update on an unmounted component.` We might need to do something like set a flag on `componentWillUnmount` so this callback can check it before trying to set state. Even better if `componentWillUnmount` can cancel this network request...?